src/System.Data.Common/src/System/Data/DbSchemaFactory.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace System.Data.Common
+{
+    public abstract class DbSchemaFactory
+    {
+        abstract public List<DbColumn> GetColumnSchema(DbDataReader dataReader);
```

Comment:
Agreed. 
@justinvp and @NickCraver Thanks for the feedback.

@justinvp I have not done a formal API review yet. I do intend to send out a request when submitting a Contracts change. I did want to expose the idea/direction to @NickCraver using code to get any feedback from him and answer questions if any. 

I will look forward for any comments from @terrajobst @KrzysztofCwalina